optimade/models/structures.py

```diff
@@ -758,6 +764,62 @@ class StructureResourceAttributes(EntryResourceAttributes):
         queryable=SupportLevel.MUST,
     )
 
+    @validator("chemical_formula_reduced", "chemical_formula_hill")
+    def check_ordered_formula(cls, v, field):
+        if v is None:
```

Comment:
This way it'll return if it's an empty string as well.
The tiniest of optimizations 😅 
```suggestion
        if not v:
```

Reply:
Do we want that though? It's actually an interesting point, an empty string will pass the regexp, but surely shouldn't be allowed?

Should the regexp be `^([A-Z][a-z]?\d*)+$` instead of `^([A-Z][a-z]?\d*)*$`?

Reply:
I don't know. I would probably consider it equivalent to `null`. But there's an argument to be made for the opposite for sure.

Reply:
Following our Slack chat, let's leave this as it is for now. When merging the schema into the spec, we can bring up whether it should permit an empty string

Reply:
Indeed, on slack we discussed this and found that we are unsure whether an empty string (`""`) should be considered a valid value or not. Something we will have to take up with in the specification repository.

Currently, however, we want to maintain the workflow of: "If value is `null`/`None`, don't do validation, otherwise _always_ validate the value".